projects/asn1-proto/asn1_proto_converter.cc

```diff
@@ -0,0 +1,134 @@
+#include "asn1_proto_converter.h"
+
+namespace asn1_proto {
+
+uint8_t ASN1ProtoConverter::GetNumBytes(size_t num) {
+  uint8_t shift = sizeof(size_t);
+  while (shift != 0) {
+    if (((num >> (shift * 8)) & 0xFF) != 0) {
+      return shift;
+    }
+    shift--;
+  }
+  return 0;
+}
+
+void ASN1ProtoConverter::AppendBytes(size_t num, size_t pos) {
+  uint8_t len_num_bytes = GetNumBytes(num);
+  std::vector<uint8_t> len_vec;
+  for (uint8_t i = len_num_bytes; i != 0; i--) {
+    len_vec.push_back((num >> (i * 7)) & 0xFF);
+  }
+  len_vec.push_back((num & 0xFF));
+  encoder_.insert(encoder_.begin() + pos, len_vec.begin(), len_vec.end());
+}
+
+size_t ASN1ProtoConverter::LongForm(const size_t assigned_len,
+                                    const size_t len_pos) {
+  uint8_t len_bytes = GetNumBytes(assigned_len);
+  if (assigned_len > 127) {
+    uint8_t longForm = (1 << 7);
+    longForm += len_bytes;
+    AppendBytes(longForm, len_pos);
+    return len_bytes;
+  }
+  return 0;
+}
+
+size_t ASN1ProtoConverter::ParseLength(const Length &len,
+                                       const size_t actual_len,
+                                       const size_t len_pos) {
+  size_t assigned_len = 0;
+  if (len.has_length_override()) {
+    sscanf(len.length_override().c_str(), "%zu", &assigned_len);
```

Comment:
We definitely *don't* want to use sscanf.

I think this might require rethinking how this function works in the broader context: the idea is that you'd directly copy the bytes from `length_override` for where the `length` would normally go.

So you'd just
```suggestion
    encoder_.insert(encoder_.begin() + len_pos, len.length_override().begin(), len.length_override().end());
    return len.length_override().size();
```

Reply:
I need to convert it to size_t because I need to eventually pass it through EncodeLongForm. If sscanf needs to be avoided, I can do something like:
```
  std::stringsteam len_read(len.length_override());
  len_read >> assigned_len;
```

Reply:
I'm not sure I understand why it needs to go through `EncodeLongForm`? I think the goal of avoiding sscanf is to try to avoid any parsing of the input; it should be something we can just splat into the buffer, like above. I'm probably missing an important detail here, we can chat about it in person :)

Reply:
Ah that's right, we're just splatting length_override in the buffer -- not trying to properly encode it. Yes, I follow and agree now.